server/src/application/application.service.ts

```diff
@@ -11,6 +12,26 @@ export class ApplicationService {
     private applicationRepository: Repository<Application>
   ) {}
 
+  async getApplications(query: GetApplicationsDto): Promise<Application[]> | null {
+    const facilityName = query.facilityName;
+    const orderBy = query.orderBy ? query.orderBy : 'facilityName';
+    const order = query.order || 'ASC';
+    const skip = query.skip || 0;
+    const take = query.take || 25;
+
+    const queryBuilder = this.applicationRepository
+      .createQueryBuilder('app')
+      .orderBy(`app.${orderBy}`, order)
+      .skip(skip)
+      .take(take);
+
+    if (facilityName) queryBuilder.andWhere('app.facilityName = :facilityName', { facilityName });
+
+    const applications = await queryBuilder.getMany();
```

Comment:
Should return the count for pagination! Refer https://github.com/EYDS-CA/magic-pin-trading/blob/develop/server/src/home/home.controller.ts#L67 end-to-end!